### ISSUE TITLE
Adding items dynamically: data-binding not works

### ISSUE THREAD

**Issue by web-padawan** [Saulis/iron-data-table#101]
I have a table (list of users) and a button with input "Add new user". 
Each row has a checkbox to change a boolean property.

After adding new user to `items` array (via `this.push`), there is a bug:
1. `item-changed` event is fired when toggling checkbox,
2. `item.admin.value` is updated,
3. but changes are not propagated to data-binding. 

**Steps to reproduce:** 
1. Open [my sandbox](https://sandbox-e0d4b.firebaseapp.com/demo/table/)
2. Add new user
3. Toggle corresponding checkbox in the table

The code is [here](https://github.com/web-padawan/datatable-sandbox/blob/master/src/components/pages/sandbox-demo-table-sub-page/sandbox-demo-table-sub-page.html).

**Comment by web-padawan** [comment]
UPD: this used to work in `beta3`.

**Comment by Saulis** [comment]
This needed some digging: `<iron-list>` uses `Polymer.Collection` to store the `items`. If you're  unfamiliar with `Polymer.Collection`, it offers helpers to map objects to indexes and vice-versa. Now, `<iron-list>` has a short-coming: it doesn't update the collection mappings when `items` is mutated, a new item is added for instance. Actually, OTOH I'm not even sure if `Polymer.Collection` has any API to update the mappings.

Anyway, I previously had a monkeypatch in `<iron-data-table>` which updated the mappings because otherwise things like selection broke down when filtering etc. I removed the monkeypatch in `beta2` after I discarded all the selection functionality from `<iron-list>` and implemented my own. 

In `beta4` I started using the `[[item]]` bindings from `<iron-list>` and as it now turned out, that feature also relies on the `Polymer.Collection` inside being up-to-date. 

Long story short, I added my old monkeypatch back and it seems to fix the issue you had, please take a look at #103.

**Comment by web-padawan** [comment]
@Saulis Thank you for detailed explanation!
Yes, #103 fixes that: updating item is propagated to data bindings inside of `data-table-column`.
